### ISSUE TITLE
Bug or documentation issue with "Instantiating with shared parameters"

### ISSUE THREAD

**Issue by michaelaye** [holoviz/param#567]
https://param.holoviz.org/user_guide/Parameters.html#instantiating-with-shared-parameters

seems to say that, by default, parameter values are independent, while the doc example itself shows how it is not independent, as changing the list from [1,2,3] to [1,2,5] was also true for the other instances in the `ss` list of `S()` instances.

The following introduction of the context manager `shared_parameters` therefore seems superfluous?

![image](https://user-images.githubusercontent.com/69774/139520517-187df7f2-e56c-4a79-b5be-d4ad9e0a3081.png)

As one can see, the list is changed for all objects in both cases?

**Comment by jbednar** [comment]
Hmm. This behavior depends on the value of `instantiate`, which differs per Parameter type. E.g. you'll get the expected results with:

```
import param

class S(param.Parameterized):
    l = param.Parameter([1,2,3], instantiate=True)

ss = [S() for i in range(10)]
ss[0].l[2]=5
ss[1].l
```

However, I'm wondering whether the behavior changed for that example, as unless I was delirious I wouldn't have written that text about that behavior.  I'll have to investigate.

**Comment by jbednar** [comment]
Ok, yes, I was delirious. Or more specifically, when I first wrote that example, I believe it used `l = param.List([1,2,3])`, which is good practice for a parameter whose value is a list, and a `param.List` automatically does `instantiate=True`. So the example worked fine originally, but then I noticed that in this section of the user guide, I had not yet introduced `param.List` or any other parameter type beyond `Parameter`, so I updated it to use `param.Parameter`, and failed to notice that the value of `instantiate` changed in the process. 

In #568 I've updated the example to force `instantiate=True` so that it illustrates what I was intending to illustrate, and I've also added more explanation of `instantiate` and how it compares to `per_instance`.  Thanks for catching this very confusing problem!
